### PR TITLE
Create user

### LINKED ISSUE #7: Account Creation
As a visitor,
I can register by filling out form,
where the app sends information
```
{
    "first_name": "First Name",
    "last_name": "Last Name",
    "phone_number": "3033033030",
    "email": "Name@email.com",
    "password": "password",
    "password_confirmation": "password"
}
``` 
to `POST /api/v1/users`,
where user will be created,
then the user's api key will be generated,
which returns the response
```
{
  "data": 
  {
    "type": "users",
    "id": "1",
    "attributes": 
    {
         "first_name": "First Name",
         "last_name": "Last Name",
         "phone_number": "3033033030",
         "email": "FirstLast@email.com",
         "api_key": "asD7fsdfwef2332!%sdf"
    }
  }
}
```
### Checks
- [x] Generates api key when creating user
- [x] Bcrypt to encrypt password
- [x] For success, returns status code 201
- [x] Returns correct response format json
- [x] Add endpoint documentation to README.md

Sad path
- [x] All fields are required for success, returns status code 422

### PR DESCRIPTION
**Pull request by lpile**
## Changes proposed in this pull request:
* Add account creation spec
* Add api_key column to users table
* Add fast_jsonapi gem
* Add #create action to UsersControllers with route
* Add api_key creation method to User model
* Add UserSerializer for JSON API Compliant
* Add create user endpoint to README.md

## Card(s) closed in this pull request:
close #7

## What did you struggle on to complete?
* Remembering how to do things in Ruby after some time off

### Test Coverage Percentage: 100%
- [x] Tested my new feature(s) as well as any feasible edge cases (if possible)
- [x] Checked coverage/index.html - did not add any new code that's not covered by testing (if possible)
- [x] Ran the test suite - all tests are passing
- [x] Merged in the latest master to my branch with git pull origin master & resolved merge conflicts
- [x] I have commented my code, particularly in hard-to-understand areas

## Helpful Resources:
* [How to add a new column to a table](https://stackoverflow.com/questions/4834809/adding-a-column-to-an-existing-table-in-a-rails-migration)
* [Fast JSONAPI docs](https://github.com/Netflix/fast_jsonapi)
* [Mod 2 Authentication Lesson](https://backend.turing.io/module2/lessons/authentication)
* [JSON API Compliant docs](https://jsonapi.org/format/)

### CODE REVIEW

File: app/serializers/user_serializer.rb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+class UserSerializer
```
Comment (annekemcgrady) [inline app/serializers/user_serializer.rb]: Logan - when we have pairing time will you give me the rundown on serializers?

Comment (lpile) [reply app/serializers/user_serializer.rb]: Of course
